core/src/main/java/com/netflix/conductor/core/execution/tasks/DoWhile.java

```diff
@@ -0,0 +1,168 @@
+	public void cancel(Workflow workflow, Task task, WorkflowExecutor executor) {
+		task.setStatus(Status.CANCELED);
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean execute(Workflow workflow, Task task, WorkflowExecutor provider) {
+		
+		boolean allDone = true;
+		boolean hasFailures = false;
+		StringBuilder failureReason = new StringBuilder();
+		List<WorkflowTask> loopOver = task.getWorkflowTask().getLoopOver();
+
+		for(WorkflowTask workflowTask : loopOver){
+			String refName = getTaskRefName(workflowTask, task);
+			Task loopOverTask = workflow.getTaskByRefName(refName);
+			if(loopOverTask == null){
+				//Task is not even scheduled yet
+				allDone = false;
+				break;
+			}
+			Status taskStatus = loopOverTask.getStatus();
+			hasFailures = !taskStatus.isSuccessful();
+			if(hasFailures){
+				failureReason.append(loopOverTask.getReasonForIncompletion()).append(" ");
+			}
+			task.getOutputData().put(workflowTask.getTaskReferenceName(), loopOverTask.getOutputData());
+			allDone = taskStatus.isTerminal();
+			if(!allDone || hasFailures){
+				break;
+			}
+		}
+		if (hasFailures) {
+			return markLoopTaskFailed(task, failureReason.toString());
+		} else if (!allDone) {
+			return false;
+		}
+		boolean shouldContinue = getEvaluatedCondition(task, workflow);
+		logger.debug("taskid {} condition evaluated to {}",task.getTaskId(), shouldContinue);
+		if (shouldContinue){
+			return scheduleLoopTasks(task, workflow, provider);
+		} else {
+			return markLoopTaskSuccess(task);
+		}
+	}
+
+	private String getTaskRefName(WorkflowTask workflowTask, Task task) {
+		String refName = workflowTask.getTaskReferenceName();
+		if (task.getIteration()>0) {
+			refName +=  "_" + task.getIteration();
+		}
+		return refName;
+	}
+
+	Map<String, Object> getConditionInput(Task task, Workflow workflow) {
+		Map<String, Object> map = new HashMap<>();
+		List<WorkflowTask> loopOverTask = task.getWorkflowTask().getLoopOver();
+		for (WorkflowTask workflowTask : loopOverTask) {
+			Task temp = workflow.getTaskByRefName(getTaskRefName(workflowTask, task));
+			map.put(workflowTask.getTaskReferenceName(), temp.getOutputData());
+			logger.debug("Taskname {} output {}", temp.getReferenceTaskName(), temp.getOutputData());
+		}
+		return map;
+	}
+
+	boolean scheduleLoopTasks(Task task, Workflow workflow, WorkflowExecutor provider) {
+		logger.debug("Scheduling loop tasks for taskid {} as condition {} evaluated to true",
+				task.getTaskId(), task.getWorkflowTask().getLoopCondition());
+		List<WorkflowTask> loopOver = task.getWorkflowTask().getLoopOver();
+		List<Task> taskToBeScheduled = new ArrayList<>();
+		int iteration = task.getIteration() + 1;
+		for(WorkflowTask loopOverRef : loopOver){
+			String refName = getTaskRefName(loopOverRef, task);
+			Task existingTask = workflow.getTaskByRefName(refName);
+			existingTask.setRetried(true);
+			Task newTask = provider.taskToBeRescheduled(existingTask);
+			newTask.setSeq(existingTask.getSeq());
```

Comment:
Why update the sequence number here, since this is auto-populated within the `scheduleTask` method?